packages/cli/src/workflows/workflow-execution.service.ts

```diff
@@ -100,12 +100,18 @@ export class WorkflowExecutionService {
 		partialExecutionVersion: 1 | 2 = 1,
 	) {
 		const pinData = workflowData.pinData;
-		const pinnedTrigger = this.selectPinnedActivatorStarter(
+		let pinnedTrigger = this.selectPinnedActivatorStarter(
 			workflowData,
 			startNodes?.map((nodeData) => nodeData.name),
 			pinData,
 		);
 
+		// if we have a trigger to start from and it's not the pinned trigger
+		// ignore the pinned trigger
+		if (pinnedTrigger && triggerToStartFrom && pinnedTrigger.name !== triggerToStartFrom.name) {
+			pinnedTrigger = null;
+		}
```

Comment:
The added tests don't test this line:
![image](https://github.com/user-attachments/assets/7fd78a0c-f6d3-457b-823d-061fae8db4ee)

Reply:
@despairblue addressed feedback. The `mock` function was causing me too much headache with sideeffects so avoided that here. Let me know if you have any more feedback